pkg/chain/bitcoin/handle.go

```diff
@@ -1,31 +1,112 @@
 package bitcoin
 
-// Handle serves as an interface abstraction around bitcoin network queries
+import (
+	"context"
+	"time"
+
+	"github.com/ipfs/go-log"
+	"github.com/keep-network/keep-ecdsa/pkg/utils"
+)
+
+var logger = log.Logger("bitcoin")
+
+const (
+	defaultTimeout             = 5 * time.Minute
+	defaultVbyteFeeFor25Blocks = int32(75)
+)
+
+type internalHandle interface {
+	broadcast(transaction string) error
+	vbyteFeeFor25Blocks() (int32, error)
+	isAddressUnused(btcAddress string) (bool, error)
+}
+
 type Handle interface {
-	Broadcast(transaction string) error
-	VbyteFeeFor25Blocks() (int32, error)
-	IsAddressUnused(btcAddress string) (bool, error)
+	Broadcast(transaction string)
+	VbyteFeeFor25Blocks() int32
+	IsAddressUnused(btcAddress string) bool
```

Comment:
Some quick thoughts:
- I think we should still return errors rather than logging in this layer. My take on the separation is “let the handle do its very best to get the right answer from the Bitcoin server, let the caller do what it wants if the handle fails after trying”.
- I wonder if we need the `offlineHandle` or even `internalHandle` at all at this juncture. I'm thinking back to my suggestion that the handle's query structure is similar across each of its three calls (each one being an HTTP request and a response or error; this is now extended by an HTTP request _with retries_ and a response or error). It feels like we can implement each `Handle` function as a wrapper that handles the HTTP request stuff, then does whatever the particular function needs to do with the response. That common wrapper can also handle `if apiUrl == "" { return ...error... }`. On the surface, this feels simpler to me now, for what is currently a fairly simple problem to solve. If we later introduce multiple handles for different server types and such that may shift, but right now it's not clear that we need it.